How-To-Setup-Internal-Package-Repository.md

```diff
@@ -271,35 +256,34 @@ Jenkins requires several PowerShell scripts to automate the processes. Create a
   . .\ConvertTo-ChocoObject.ps1
 
   # get all of the packages from the test repo
-  $testPkgs = choco list --source $TestRepo | Select-Object -Skip 1 | Select-Object -SkipLast 1 | ConvertTo-ChocoObject
-  $prodPkgs = choco list --source $ProdRepo | Select-Object -Skip 1 | Select-Object -SkipLast 1 | ConvertTo-ChocoObject
+  $testPkgs = choco list --source $TestRepo -r | ConvertTo-ChocoObject
+  $prodPkgs = choco list --source $ProdRepo -r | ConvertTo-ChocoObject
   $tempPath = Join-Path -Path $env:TEMP -ChildPath ([GUID]::NewGuid()).GUID
+
   if ($null -eq $testPkgs) {
       Write-Verbose "Test repository appears to be empty. Nothing to push to production."
   }
+
   elseif ($null -eq $prodPkgs) {
       $pkgs = $testPkgs
   }
+
   else {
       $pkgs = Compare-Object -ReferenceObject $testpkgs -DifferenceObject $prodpkgs -Property name, version | Where-object SideIndicator -eq '<='
   }
 
   $pkgs | ForEach-Object {
       Write-Verbose "Downloading package '$($_.name)' to '$tempPath'."
-      choco download $_.name --no-progress --output-directory=$tempPath --source=$TestRepo
-
+      choco download $_.name --no-progress --output-directory=$tempPath --source=$TestRepo --ignore-dependencies

```

Comment:
This actually changes the functionality of the script - why did you add `--ignore-dependencies`?

Reply:
I believe the logic here is that the other scripts will have pulled down and internalized those dependencies, and having `--ignore-dependencies` here will keep choco from re-downloading the potential same dependencies repeatedly, which would require additional error handling/cleanup between packages.

Reply:
If the dependencies for a package ever change then this won't be picked up by anything other than this script so at that stage you potentially have a package either failing to install (as it can't find it's new dependency) or it reached out to chocolatey.org for that dependency. So that needs to be balanced. I appreciate the chances of a package dependency being added and it not being one of the 'common ones' (that is likely to already be internalized) is rare, but it does happen.

The error handling is already in the code (it captures the exit code at line 291 below and uses a Write-Verbose to output the message). Jenkins _shouldn't_ fail the job in this case as there are no terminating errors from the PowerShell. I say _shouldn't_ and I also say _whenever I've used this is demos and talks (which is a lot) I don't remember any problems_ but it should be tested.

Reply:
I read that code back just now and had my `ah-ha` moment. Ryan has just pushed a commit  to revert this.